src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs

```diff
@@ -4513,9 +4513,15 @@ internal bool TryReadSqlValue(SqlBuffer value, SqlMetaDataPriv md, int length, T
 
         private bool TryReadSqlDateTime(SqlBuffer value, byte tdsType, int length, byte scale, TdsParserStateObject stateObj)
         {
+#if netcoreapp
+            Span<byte> datetimeBuffer = length<=10 ?  stackalloc byte[length] : new byte[length];
```

Comment:
I am just curious - why 10?

Reply:
Later on in the method there is a switch and asserts for each of the valid date types, 10 is the largest number of bytes needed by any of those cases. There is the possibility that the method can be called with an invalid type or unusual buffer length and if those happen the old heap based behaviour will be used.

Reply:
should it be a constant?